elastic/kibana
# [Embeddable rebuild] Allow Dashboard to provide references

## Summary
This PR fixes https://github.com/elastic/kibana/issues/176229 by adding the ability for the Dashboard to provide references for its React Embeddable children to inject, and adding the ability for the React Embeddable children to provide extracted references back to the Dashboard.

Also adds a Field List React embeddable example which references a selected Data view. This can be used to test the behaviours of this PR.

![Field List](https://github.com/elastic/kibana/assets/14276393/8e6e059c-2810-4df2-a15c-c38626eed8b1)


## How to test
- run Kibana with `yarn start --run-examples`.
- Create a Dashboard and add a `Field List` embeddable.
- Select a data view
- Save the Dashboard
- In the saved object management page, ensure the Dashboard has a reference to the selected Data view.
- Open the Dashboard, ensuring that the data view is properly injected, and that there are no `unsaved changes`.

<img width="1070" alt="Screenshot 2024-02-08 at 12 45 39 PM" src="https://github.com/elastic/kibana/assets/14276393/d553113a-ac6d-44cb-a7a3-e270fb6ff0ea">
